middleware-catalog-module/catalog-model/src/main/java/co/tpg/catalog/model/Subject.java

```diff
@@ -1,4 +1,37 @@
 package co.tpg.catalog.model;
 
-public class Subject {
-}
+import lombok.*;
+
+/**
+ * Response class for Subject entity
+ * @author maxx
+ * @since 2019-10-03
+ */
+
+
+@Builder
+@Setter
+@Getter
+@ToString
+@EqualsAndHashCode
+public class Subject extends AbstractModel<String> {
+    private String id;
```

Comment:
You need to use the annotation @EqualsAndHashCode.Exclude to remove the attributes that are not part of equals and hashCode method. Look at TeachingClass.java for example.